src/shared/remoteSession.ts

```diff
@@ -173,13 +180,47 @@ export async function handleMissingTool(tools: Err<MissingTool[]>) {
 export async function getDeniedSsmActions(client: IamClient, roleArn: string): Promise<IAM.EvaluationResult[]> {
     const deniedActions = await client.getDeniedActions({
         PolicySourceArn: roleArn,
-        ActionNames: [
-            'ssmmessages:CreateControlChannel',
-            'ssmmessages:CreateDataChannel',
-            'ssmmessages:OpenControlChannel',
-            'ssmmessages:OpenDataChannel',
-        ],
+        ActionNames: minimumSsmActions,
     })
 
     return deniedActions
 }
+
+export async function promptToAddPolicies(client: IamClient, roleArn: string): Promise<boolean> {
+    const promptText = `In order to connect, toolkit will add actions ${getFormattedSsmActions()} to role ${roleArn}`
+    const confirmation = await showConfirmationMessage({ prompt: promptText, confirm: 'Approve' })
+
+    if (confirmation) {
+        await addSsmActionsToInlinePolicy(client, roleArn)
```

Comment:
If you _don't_ `await` here, could immediately return and the `Adding Inline Policy...` progress message could display sooner. 

With the current approach, there's an invisible delay and then the progress message shows later?

Reply:
I like that idea. I just tried it out and it is noticeably much smoother.